### ISSUE TITLE
[a11y]: Screen Reader doesn't read options in Dropdown correctly

### ISSUE THREAD

**Issue by RachelGHogan** [carbon-design-system/carbon#17582]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.64.1 (reproduced in 1.66.0)

### Automated testing tool and ruleset

Screenreader

### Assistive technology

Google Chrome Screen Reader v53.0.2784.15

### Description

It seems that if you have multiple properties within an object that you pass through the items field (even if one of them matches the itemToString function), all the screenreader says is "expanded", instead of the expected "[option name/ abel] not selected x of y". 

You can see in the below screenshot that the itemToString is parsing just fine, since the label of the dropdown is option 3, and you can see in the console as well that both option 1 and 3 were correctly parsed. 
![image](https://github.com/user-attachments/assets/5ae024ac-fbab-428c-908f-3a6b549fbc03)

### Reproduction/example

https://stackblitz.com/edit/github-ba7pgw-j28fwm?file=src%2FApp.jsx

### Steps to reproduce

Pass a list with an object with more than 1 property to the items field of a dropdown. Example available in the stackblitx above

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by finken2** [comment]
I'm running into the same thing. 
We have two dropdowns on a page: 
 - one dropdown which works fine
 - another which won't read the items
I believe this issue isn't multiple properties, but actually the **id** property being in the item. 

I took your stackblitz and modified/simplified a little: https://stackblitz.com/edit/github-ba7pgw-xpupkw?file=src%2FApp.jsx
Here's a recording as well, with macos voiceover (but no audio)
https://github.com/user-attachments/assets/a6612312-7426-4265-a623-fc62da7b4f84

I also tested in our code on ibm cloud, just deleting the `id` property from a list of items, and it seems to resolve the issue.

**Comment by finken2** [comment]
I think i got to the bottom of this today:
The root cause is the `aria-activedescendant` in downshift's menu updates to the IDs of the different items in the list. That's fine when downshift is managing the IDs, which it is if the items don't contain IDs.
However, if your items have IDs, that's what ends up on the DOM node b/c of [this PR in carbon](https://github.com/carbon-design-system/carbon/pull/15249), so the downshift aria info is pointing to nonexistent items, and nothing is read.  I think the solution is to rollback that PR, unless we're going to get into the downshift code. 

TLDR: Basically, downshift is using the ID for things, so we shouldn't be mucking with it.

**Comment by finken2** [comment]
PR mentioned by andrew on slack that should have fixed this issue, but seems like it didn't:
https://github.com/carbon-design-system/carbon/pull/16688

**Comment by guidari** [comment]
There is an issue open in Downshift related to that as well
https://github.com/downshift-js/downshift/issues/1607

**Comment by emyarod** [comment]
yeah it looks like #16688 resolves the issue, but just for the Storybook example component. As @guidari mentioned, this is an issue found in Downshift. We will have to evaluate how to proceed from here so that we aren't stuck with this upstream blocker

**Comment by kzhang-2017** [comment]
Hi. Any updates on this issue?

**Comment by valZho** [comment]
I'm not sure if this is related or not, but I do notice another issue with Dropdown that seems like it could be a possible culprit (and should be addressed either way)...

I'm currently seeing Dropdown creating something like the following structure:

```
<div class="cds--dropdown__wrapper cds--list-box__wrapper">
  <label class="cds--label" id="downshift-:rd:-label" for="downshift-:rd:-toggle-button">Select Device</label>
  <div id="MY-ID" class="cds--dropdown cds--list-box">
    <button
      type="button"
      class="cds--list-box__field"
      title="MY TITLE"
      aria-activedescendant=""
      aria-controls="downshift-:rd:-menu"
      aria-expanded="false"
      aria-haspopup="listbox"
      id="downshift-:rd:-toggle-button"
      role="combobox"
      tabindex="0"
      aria-label="MY TITLE"
    >
      <span class="cds--list-box__label">MY LABEL</span>
      <div class="cds--list-box__menu-icon">
        <svg
          focusable="false"
          preserveAspectRatio="xMidYMid meet"
          fill="currentColor"
          name="chevron--down"
          aria-label="Open menu"
          width="16" height="16"
          viewBox="0 0 16 16"
          role="img"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"></path>
          <title>Open menu</title>
        </svg>
      </div>
    </button>
    <ul
      id="downshift-:rd:-menu"
      class="cds--list-box__menu"
      role="listbox"
      aria-labelledby="downshift-:rd:-label"
    >
    </ul>
  </div>
</div>
```

The `for` and `id` properties for the Dropdown are NOT set to use/include the user-supplied ID. This means that if someone is using more than one Dropdown on a page, you are going to run into ID conflicts. ... I'm not sure if https://github.com/carbon-design-system/carbon/pull/16688 fixes this or not.

**Comment by tay1orjones** [comment]
A common violation (#18858,18855) from this is `The 'id' "downshift-:r2:-label" specified for the ARIA property 'aria-labelledby' value is not valid`

I'm going to bump up the severity of this.
